gbif/diagnostics/src/main/java/org/gbif/pipelines/diagnostics/RepairGbifIDLookupTool.java

```diff
@@ -0,0 +1,187 @@
+package org.gbif.pipelines.diagnostics;
+
+import com.beust.jcommander.JCommander;
+import com.beust.jcommander.Parameter;
+import java.io.File;
+import java.util.Set;
+import javax.validation.constraints.NotNull;
+import lombok.Builder;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.hadoop.hbase.client.Connection;
+import org.gbif.dwc.Archive;
+import org.gbif.dwc.DwcFiles;
+import org.gbif.dwc.record.Record;
+import org.gbif.dwc.terms.DwcTerm;
+import org.gbif.pipelines.diagnostics.strategy.DeletionStrategy.DeletionStrategyType;
+import org.gbif.pipelines.keygen.HBaseLockingKeyService;
+import org.gbif.pipelines.keygen.common.HbaseConnectionFactory;
+import org.gbif.pipelines.keygen.config.KeygenConfig;
+import org.gbif.pipelines.keygen.identifier.OccurrenceKeyBuilder;
+
+@Slf4j
+@Builder
+public class RepairGbifIDLookupTool {
+
+  @Parameter(names = "--datase-key", description = "GBIF registry ID for the dataset")
+  @NotNull
+  public String datasetKey;
+
+  @Parameter(
+      names = "--input-directory",
+      description = "Use DWCA archive as a lookup keys source, provide full input path")
+  public File dwcaDirectory;
```

Comment:
Should this be named dwcaSource? It seems to support both files and directories.